### PR TITLE
Upgrade ember cli to 3.7.1

### PR DESCRIPTION
**Pull request by patocallaghan**
To remove a ember-cli-babel 5.x deprecation warning in my Ember CLI terminal I've upgraded the addon to run on Ember CLI 3.7. All but one file (addon/index.js) are from the default Ember CLI upgrade

```
DEPRECATION: ember-cli-babel 5.x has been deprecated. Please upgrade to at least ember-cli-babel 6.6. Version 5.2.8 located: embercom -> embercom-computed-properties -> ember-string-fmt -> ember-cli-babel
```

One change that needed to be made was that since Ember 3.x `Ember.inspect` now adds quotes to strings past to it.

```js
// 1.x - 2.x
let foo = "bar";
Ember.inspect(foo) //=> bar

// 3.x
let foo = "bar";
Ember.inspect(foo) //=> "bar"
```

2.18: [Twiddle](https://ember-twiddle.com/42e540deee5046cf3ed00a5ddba4a375)
3.4: [Twiddle](https://ember-twiddle.com/82c087503fb61d540ff2490ec02992e2)

To workaround this issue I've opted to not use `Ember.inspect` if the variable passed is a string.

### CODE REVIEW

File: addon/index.js [hunk 1]
```diff
@@ -19,6 +17,7 @@ export default function(str, formats) {
   return str.replace(/%@([0-9]+)?/g, function(s, argIndex) {
     argIndex = (argIndex) ? parseInt(argIndex, 10) - 1 : idx++;
     s = cachedFormats[argIndex];
-    return (s === null) ? '(null)' : (s === undefined) ? '' : inspect(s);
+    s = typeof s === 'string' ? s : inspect(s);
```
Comment (patocallaghan) [inline addon/index.js:20]: These are the only changes required to get the test suite to pass. Only pass non-strings to `Ember.inspect` as its behaviour changed between 1.x/2.x and 3.x.

**Comment by patocallaghan** [comment]
CI is 🔴, looking into it

**Comment by patocallaghan** [comment]
Failing test in CI

image

In Ember 2.x hashes get rendered like `{id: 3}` and in 3.x they get rendered like `{ id: 3 }`

File: tests/unit/fmt-test.js [hunk 2]
```diff
@@ -16,7 +17,11 @@ test('\'%@08 %@07 %@06 %@05 %@04 %@03 %@02 %@01\'.fmt(\'One\', \'Two\', \'Three\
 });
 
 test('\'data: %@\'.fmt({ id: 3 }) => \'data: {id: 3}\'', function(assert) {
-  assert.equal(fmt('data: %@', [{ id: 3 }]), 'data: {id: 3}');
+  // The whitespace output of hashes passed through Ember.inspect differs from 3.x. We need have different output whether it's 2.x or 3.x
+  let emberVersion = /\d/.exec(VERSION);
+  let emberInt = parseInt(emberVersion[0], 10);
+  let hashOutput = emberInt > 2 ? '{ id: 3 }' : '{id: 3}'
+  assert.equal(fmt('data: %@', [{ id: 3 }]), `data: ${hashOutput}`);
```
Comment (patocallaghan) [inline tests/unit/fmt-test.js:24]: When running tests through ember-try, i.e. using Ember 2.x and 3.x, we need to tweak the output of the test depending what version of Ember we are running.

Ember 2.x: `{id: 3}`
Ember 3.x: `{ id: 3 }`

**Comment by patocallaghan** [comment]
CI is now 🍏

**Review by GavinJoyce — APPROVED** [review]

**Comment by GavinJoyce** [comment]
@patocallaghan I've added you as a collaborator on npm: https://www.npmjs.com/package/ember-string-fmt

**Comment by patocallaghan** [comment]
n1 @GavinJoyce thanks :bowing_man: We should create 1.0 release with the new changes

**Pull request merged**